### PR TITLE
electron-mail.profile refactoring

### PR DESCRIPTION
**Pull request by pirate486743186**
title.

~~noroot is missing from all the chromium browsers that i checked and electron the electron sdk....~~

Now refactoring with electron.profile

### CODE REVIEW

**Review by rusty-snake — CHANGES_REQUESTED** [review]
I'm against removing `noroot` because of portals, the fix needs to be something else.

However in the case of electron-mail it should be an electron redirect profile to fix any potential electron problematic.

File: etc/profile-a-l/electron-mail.profile [hunk 1]
```diff
@@ -6,50 +6,35 @@ include electron-mail.local
 # Persistent global definitions
 include globals.local
 
+ignore noblacklist ${HOME}/.config/Electron
+ignore noblacklist ${HOME}/.config/electron*-flag*.conf
+ignore whitelist ${HOME}/.config/Electron
+ignore whitelist ${HOME}/.config/electron*-flag*.conf
```
Comment (rusty-snake) [inline etc/profile-a-l/electron-mail.profile:12]:
```suggestion
```

File: etc/profile-a-l/electron-mail.profile [hunk 2]
```diff
@@ -6,50 +6,35 @@ include electron-mail.local
...
 mkdir ${HOME}/.config/electron-mail
 whitelist ${HOME}/.config/electron-mail
-whitelist ${DOWNLOADS}
 
-include whitelist-common.inc
-include whitelist-runuser-common.inc
-include whitelist-usr-share-common.inc
-include whitelist-var-common.inc
+# Add the next line to your electron.local if your kernel allows unprivileged users clone.
+#include chromium-common-hardened.inc.profile
```
Comment (rusty-snake) [inline etc/profile-a-l/electron-mail.profile:24]:
```suggestion
```

File: etc/profile-a-l/electron-mail.profile [hunk 3]
```diff
@@ -6,50 +6,35 @@ include electron-mail.local
...
 
-apparmor
 caps.drop all
```
Comment (rusty-snake) [inline etc/profile-a-l/electron-mail.profile:26]:
```suggestion
```

File: etc/profile-a-l/electron-mail.profile [hunk 4]
```diff
@@ -6,50 +6,35 @@ include electron-mail.local
...
-netfilter
 no3d
-nodvd
-nogroups
-noinput
 nonewprivs
```
Comment (rusty-snake) [inline etc/profile-a-l/electron-mail.profile:28]:
```suggestion
```

File: etc/profile-a-l/electron-mail.profile [hunk 5]
```diff
@@ -6,50 +6,35 @@ include electron-mail.local
...
-noroot
-notv
-nou2f
-novideo
 protocol unix,inet,inet6,netlink
 seccomp !chroot
```
Comment (rusty-snake) [inline etc/profile-a-l/electron-mail.profile:30]:
```suggestion
```

File: etc/profile-a-l/electron-mail.profile [hunk 6]
```diff
@@ -6,50 +6,35 @@ include electron-mail.local
...
-# tracelog - breaks on Arch
+# tracelog - breaks on Arch and Debian
```
Comment (rusty-snake) [inline etc/profile-a-l/electron-mail.profile:31]:
```suggestion
```

Breaks everywhere

File: etc/profile-a-l/electron-mail.profile [hunk 7]
```diff
@@ -6,50 +6,35 @@ include electron-mail.local
 # Persistent global definitions
 include globals.local
 
+ignore noblacklist ${HOME}/.config/Electron
+ignore noblacklist ${HOME}/.config/electron*-flag*.conf
+ignore whitelist ${HOME}/.config/Electron
+ignore whitelist ${HOME}/.config/electron*-flag*.conf
+# breaks tray functionality
+ignore dbus-user none
+
```
Comment (rusty-snake) [inline etc/profile-a-l/electron-mail.profile:15]: TODO: `ignore disable-mnt` yes no?

File: etc/profile-a-l/electron-mail.profile [hunk 8]
```diff
@@ -6,50 +6,35 @@ include electron-mail.local
...
-private-cache
-private-dev
 private-etc alternatives,ca-certificates,crypto-policies,fonts,gtk-2.0,gtk-3.0,ld.so.cache,ld.so.preload,nsswitch.conf,pki,resolv.conf,selinux,ssl,xdg
 private-opt ElectronMail
-private-tmp
 
-# breaks tray functionality
-# dbus-user none
-dbus-system none
+# memory-deny-write-execute - breaks on Arch and Debian
```
Comment (rusty-snake) [inline etc/profile-a-l/electron-mail.profile:37]:
```suggestion
```

Breaks everywhere

File: etc/profile-m-z/makepkg.profile [hunk 9]
```diff
@@ -1,4 +1,5 @@
 # Firejail profile for makepkg
+# Description: A simplicity-focused packaging tool for Arch archives
```
Comment (kmk3) [inline etc/profile-m-z/makepkg.profile:2]: This change is from #5299; please rebase and remove the makepkg commit.

**Review by kmk3 — DISMISSED** [review]

File: etc/profile-a-l/electron-mail.profile [hunk 10]
```diff
@@ -1,55 +1,27 @@
...
-# Description: Unofficial desktop app for several E2E encrypted email providers
+# Description: Unofficial desktop app for the Proton Mail E2E encrypted email provider
 # This file is overwritten after every install/update
 # Persistent local customizations
 include electron-mail.local
 # Persistent global definitions
 include globals.local
 
+# breaks tray functionality
+ignore dbus-user none
```
Comment (kmk3) [inline etc/profile-a-l/electron-mail.profile:10]: See `allow-tray` in firejail.config and `?ALLOW_TRAY` in profiles.

**Comment by rusty-snake** [comment]
Can you split the refactoring and the fixes into seperate commits.

File: etc/profile-a-l/electron-mail.profile [hunk 11]
```diff
@@ -1,55 +1,39 @@
...
-novideo
-protocol unix,inet,inet6,netlink
-seccomp !chroot
-# tracelog - breaks on Arch
-
-private-bin electron-mail
-private-cache
-private-dev
+
+private-bin basename,cut,electron-mail,gio,grep,mawk,readlink,sh,tr,xdg-open,xdg-settings
```
Comment (rusty-snake) [inline etc/profile-a-l/electron-mail.profile:28]: How did you generated this list?

File: etc/profile-a-l/electron-mail.profile [hunk 12]
```diff
@@ -1,55 +1,39 @@
...
-whitelist ${DOWNLOADS}
 
-include whitelist-common.inc
-include whitelist-runuser-common.inc
-include whitelist-usr-share-common.inc
-include whitelist-var-common.inc
+# These lines are needed to allow Firefox to open links
+noblacklist ${HOME}/.mozilla
+whitelist ${HOME}/.mozilla/firefox/profiles.ini
+read-only ${HOME}/.mozilla/firefox/profiles.ini
```
Comment (rusty-snake) [inline etc/profile-a-l/electron-mail.profile:24]: Did you tested this?

File: etc/profile-a-l/electron-mail.profile [hunk 13]
```diff
@@ -1,55 +1,39 @@
...
 
-include disable-common.inc
-include disable-devel.inc
-include disable-exec.inc
-include disable-interpreters.inc
-include disable-programs.inc
-include disable-shell.inc
-include disable-xdg.inc
+# sh is needed to allow Firefox to open links
+#include disable-shell.inc
```
Comment (rusty-snake) [inline etc/profile-a-l/electron-mail.profile:16]: The use `allow-bin-sh.inc`

File: etc/profile-a-l/electron-mail.profile (hunk 11 above)
Comment (rusty-snake) [reply etc/profile-a-l/electron-mail.profile:28]: That does not work. `--build` will only grab program for your environment. Either add all required programs or drop private-bin.

File: etc/profile-a-l/electron-mail.profile (hunk 12 above)
Comment (rusty-snake) [reply etc/profile-a-l/electron-mail.profile:24]: What's your environment? Did you tested it with the profile as it is now?

File: etc/profile-a-l/electron-mail.profile (hunk 13 above)
Comment (rusty-snake) [reply etc/profile-a-l/electron-mail.profile:16]: What do you mean? What is backwards.

Comment (rusty-snake) [reply etc/profile-a-l/electron-mail.profile:16]: > The allow needs fixing?

Yes.

Comment (rusty-snake) [reply etc/profile-a-l/electron-mail.profile:16]: What does `sh --version` output for you?

Do you get any error?

File: etc/profile-a-l/electron-mail.profile [hunk 14]
```diff
@@ -1,55 +1,43 @@
 # Firejail profile for electron-mail
-# Description: Unofficial desktop app for several E2E encrypted email providers
+# Description: Unofficial desktop app for the Proton Mail E2E encrypted email provider
 # This file is overwritten after every install/update
 # Persistent local customizations
 include electron-mail.local
 # Persistent global definitions
 include globals.local
 
+# breaks tray functionality
```
Comment (rusty-snake) [inline etc/profile-a-l/electron-mail.profile:9]:
```suggestion
```

**Review by rusty-snake — APPROVED** [review]
Changes as I could see (since OP does not provided one and everything is done in one ugly commit)

- Refactor as electron redirect
- change comment
- allow some D-Bus
- removed no3d
- allow sh
- allow firefox' profiles.ini
- block sound
- harden private-etc
- drop private-bin

**Comment by kmk3** [comment]
So I thought that electron-mail.profile was something like a common profile,
but apparently it's just for a program with a very generic name.

This is the project page for "ElectronMail":

* <https://github.com/vladimiry/ElectronMail>

It is an Electron-based email client, intended for email providers that serve
email primarily through web-based protocols (and that have limited to no
support for standard email protocols), and it used to support Proton Mail and
Tutanota.

And then it dropped support for Tutanota, making it specific to Proton Mail.

While Tutanota has its own official Electron-based email client:

* <https://github.com/tutao/tutanota>

Which is not related to ElectronMail as far as I can tell.

Anyway, it was very confusing when reviewing this PR and #5325, mostly due to
how generic the names "electron-mail" and "ElectronMail" are.

For completeness, Proton Mail does not have an official desktop client, but
there are "bridges" that allow talking to the service using standard email
protocols and clients:

* Official: <https://github.com/ProtonMail/proton-bridge>
* Unofficial: <https://github.com/emersion/hydroxide>

Though note that Proton Mail only supports using bridges on paid accounts.

For reference, electron-mail.profile was added on commit 20fa0d580 ("Add new
electron-mail profile (#3053)", 2019-11-23) and tutanota-desktop.profile was
added on commit ef5c6c3f9 ("new profile: tutanota-desktop (#3870)",
2021-01-05).

File: etc/profile-a-l/electron-mail.profile [hunk 15]
```diff
@@ -1,55 +1,43 @@
 # Firejail profile for electron-mail
```
Comment (kmk3) [inline etc/profile-a-l/electron-mail.profile:1]:
```suggestion
# Firejail profile for ElectronMail
```

Clarify that there is an actual program called "ElectronMail".

Comment (netblue30) [reply etc/profile-a-l/electron-mail.profile:1]: It is kind of confusing. I think tutanota and proton started with the same code and are running now slightly different flavors of it. I would say let's make a new ElectronMail.profile and redirect it to electron-mail.profile, the same way we do it with .gimp-2.10.profile and gimp.profile. We'll fix it when people start complaining.

Comment (rusty-snake) [reply etc/profile-a-l/electron-mail.profile:1]: > electron-mail is the actual command for the app. Not ElectronMail

Yes, hence the filename is `electron-mail.profile`. The name used at user-facing places (e.g. .desktop files) is ElectronMail AFAIK. Which is IMHO the better name for this place but rarely used.

**Comment by netblue30** [comment]
@pirate486743186, @rusty-snake, @kmk3   - do we check in, is it done? Thanks!

**Comment by kmk3** [comment]
@netblue30 commented [on Oct 4](https://github.com/netblue30/firejail/pull/5298/files#issuecomment-1267273729):

> do we check in, is it done? Thanks!

Thanks for checking in; I wasn't too sure about the state of this PR either.

@pirate486743186 Could you squash the commits?

And make the commit message something like this (based on @rusty-snake's
[list][1]):

```
electron-mail.profile: refactor and redirect to electron.profile

Changes:

- redirect to electron.profile
- fix program name
- update program description
- allow /bin/sh
- allow opening links in Firefox
- remove no3d, nonewprivs, noroot, protocol, seccomp
- add machine-id, nosound
- remove private-bin, disable-mnt
- harden private-etc
- allow D-Bus notifications, secrets
```

Other than that, LGTM.

[1]: https://github.com/netblue30/firejail/pull/5298#pullrequestreview-1095690567

**Pull request merged**